cpp/src/arrow/array/concatenate.cc

```diff
@@ -296,6 +301,16 @@ class ConcatenateImpl {
     return buffers;
   }
 
+  // Gather the index-th buffer of each input into a vector.
```

Comment:
Do you still use this overload? It looks like it could just be replaced by `Buffers(size_t index, int byte_width)`

Reply:
Yes, it's still used.